src/ebonite/build/docker.py

```diff
@@ -89,6 +91,19 @@ def get_runner(self):
         return self.default_runner
 
 
+    # TODO not sure about signature, maybe pass provider here. also, need to separate kwargs for provider, for builder, and for builder kwargs
```

Comment:
For kwargs I recommend to use `DockerRunner` pattern: client passes opaque kwargs, but method receives concrete documented args and extra kwargs as "trash bin" for unneeded ones. This way we are not required to modify kwargs along call hierarchy which creates coupling through knowledge of arguments semantics.